Libplanet/Net/NetMQTransport.cs

```diff
@@ -0,0 +1,1011 @@
+
+            if (_host != null && _listenPort is int listenPortAsInt)
+            {
+                _endPoint = new DnsEndPoint(_host, listenPortAsInt);
+            }
+
+            _iceServers = iceServers?.ToList();
+            if (_host == null && (_iceServers == null || !_iceServers.Any()))
+            {
+                throw new ArgumentException(
+                    $"Swarm requires either {nameof(host)} or " +
+                    $"{nameof(iceServers)}."
+                );
+            }
+
+            _logger = logger;
+
+            _requests = new AsyncCollection<MessageRequest>();
+            _runtimeCancellationTokenSource = new CancellationTokenSource();
+            _runtimeProcessor = Task.Factory.StartNew(
+                () =>
+                {
+                    // Ignore NetMQ related exceptions during NetMQRuntime.Dispose() to stabilize
+                    // tests
+                    try
+                    {
+                        using (var runtime = new NetMQRuntime())
+                        {
+                            var workerTasks = new Task[workers];
+
+                            for (int i = 0; i < workers; i++)
+                            {
+                                workerTasks[i] = ProcessRuntime(
+                                    _runtimeCancellationTokenSource.Token
+                                );
+                            }
+
+                            runtime.Run(workerTasks);
+                        }
+                    }
+                    catch (NetMQException e)
+                    {
+                        _logger.Error(
+                            e,
+                            $"NetMQException occurred in {nameof(_runtimeProcessor)}."
+                        );
+                    }
+                    catch (ObjectDisposedException e)
+                    {
+                        _logger.Error(
+                            e,
+                            $"ObjectDisposedException occurred in {nameof(_runtimeProcessor)}."
+                        );
+                    }
+                },
+                CancellationToken.None,
+                TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
+                TaskScheduler.Default
+            );
+
+            Protocol = new KademliaProtocol(
+                this,
+                _privateKey.PublicKey.ToAddress(),
+                _appProtocolVersion,
+                _logger,
+                tableSize,
+                bucketSize);
+        }
+
+        /// <summary>
+        /// The <see cref="EventHandler" /> triggered when the different version of
+        /// <see cref="Peer" /> is discovered.
+        /// </summary>
+        private event EventHandler<DifferentProtocolVersionEventArgs>
+            DifferentVersionPeerEncountered;
+
+        /// <summary>
+        /// The <see cref="EventHandler" /> triggered when a <see cref="Message"/> is
```

Comment:
It's just question, is there some rules to add space before the end?

```
        /// The <see cref="EventHandler" /> triggered when a <see cref="Message"/> is
                                        ^ here                                 ^ but not this 🤔 
```

Reply:
I guess not...

Reply:
Both are valid as it's XML, but I prefer no space, because we usually have no enough horizontal space.

Reply:
I found that existing xml documents also have this problem. This will be great to be dealt in new PR .